File: ui/actiondisplay/icons.go
package actiondisplay

import (
	"Sqyre/internal/assets"
	"Sqyre/internal/models/actions"

	"fyne.io/fyne/v2"
	"fyne.io/fyne/v2/theme"
)

func Icon(action actions.ActionInterface) fyne.Resource {
	switch a := action.(type) {
	case *actions.Click:
		if a.State {
			return assets.MouseClickFilledIcon
		}
		return assets.MouseClickIcon
	case *actions.Move:
		return assets.MouseIcon
	case *actions.Key:
		if a.State {
			return theme.DownloadIcon()
		}
		return theme.UploadIcon()
	case *actions.Type:
		return theme.DocumentIcon()
	case *actions.Wait:
		return theme.HistoryIcon()
	case *actions.Pause:
		return theme.MediaPauseIcon()
	case *actions.FocusWindow:
		return theme.VisibilityIcon()
	case *actions.RunMacro:
		return theme.MediaPlayIcon()
	case *actions.Conditional:
		return theme.QuestionIcon()
	case *actions.Loop:
		return theme.ViewRefreshIcon()
	case *actions.Break:
		return theme.MediaStopIcon()
	case *actions.Continue:
		return theme.MediaSkipNextIcon()
	case *actions.SetVariable:
		return assets.VariableIcon
	case *actions.SaveVariable:
		return theme.DocumentSaveIcon()
	case *actions.ForEachRow:
		return theme.ListIcon()
	case *actions.Ocr:
		return assets.TextSearchIcon
	case *actions.ImageSearch:
		return assets.ImageSearchIcon
	case *actions.FindPixel:
		return theme.ColorChromaticIcon()
	default:
		return theme.ErrorIcon()
	}
}

File: crates/sqyre-domain/src/display/icons.rs
//! Action type icon glyphs for the macro tree.

use crate::{Action, ActionKind};

pub fn action_icon_glyph(action: &Action) -> &'static str {
    match &action.kind {
        ActionKind::Click { state, .. } => {
            if *state {
                "⬇"
            } else {
                "⬆"
            }
        }
        ActionKind::Move { .. } => "➔",
        ActionKind::Key { state, .. } => {
            if *state {
                "⬇"
            } else {
                "⬆"
            }
        }
        ActionKind::Type { .. } => "⌨",
        ActionKind::Wait { .. } => "⏱",
        ActionKind::Pause { .. } => "⏸",
        ActionKind::FocusWindow { .. } => "👁",
        ActionKind::RunMacro { .. } => "▶",
        ActionKind::Conditional { .. } => "?",
        ActionKind::Loop { .. } | ActionKind::While { .. } => "↻",
        ActionKind::Break => "⏹",
        ActionKind::Continue => "⏭",
        ActionKind::SetVariable { .. } => "x",
        ActionKind::SaveVariable { .. } => "💾",
        ActionKind::ForEachRow { .. } => "☰",
        ActionKind::Ocr { .. } => "🔤",
        ActionKind::ImageSearch { .. } => "🔍",
        ActionKind::FindPixel { .. } => "🎨",
        ActionKind::NavigateSelect(_) => "⌖",
        ActionKind::NavigateKey { .. } => "⎇",
    }
}

/// True when the pill value looks like a `${name}` / `{name}` var ref.
pub fn looks_like_var_ref(text: &str) -> bool {
    let t = text.trim();
    (t.starts_with("${") && t.ends_with('}'))
        || (t.starts_with('{') && t.ends_with('}') && !t.starts_with("${"))
}
